### ISSUE TITLE
Basename of .html generated from symlinked markdown contains absolute path - regression in 1.9.2

### ISSUE THREAD

**Issue by dirk-thomas** [doxygen/doxygen#9294]
**Describe the bug**

The basename of generated `.html` files from markdown contains the canonical path of input files when symlinks are being used. That not only discloses information about the prefix where the doxygen invocation is happening but also affects all links between markdown pages. The regression was introduced in version 1.9.2 - specifically in this change: https://github.com/doxygen/doxygen/commit/3204a2206aa617b41c6da5999c3b826057d274f3#diff-ccf6a204f87f1890f62c4ff23332fbd6cde61daefd4b2a6a8fab3d7fec49fb91R2742

**Expected behavior**

The same `.html` filename should be generated when the input file is a file or a symlink. This was the case in version 1.9.1 and earlier.

**To Reproduce**

Run the following sequence of commands for a SSCCE:

```
mkdir /tmp/doxygen_sscce
touch /tmp/doxygen_sscce/test.md
mkdir /tmp/doxygen_sscce2
touch /tmp/doxygen_sscce2/test2.md
ln -s /tmp/doxygen_sscce2/test2.md /tmp/doxygen_sscce/test2.md
cd /tmp/doxygen_sscce
doxygen -g doxyfile
doxygen doxyfile
```

In 1.9.1 and earlier the generated file is named `html/md_test2.html`.

In 1.9.2 and later the generated file is named `html/md__tmp_doxygen_sscce2_test2.html`.

**Additional context**

In the above referenced change the code was refactored from `QFileInfo(...).absFilePath()` to `FileInfo(...).absFilePath()` but the latter function is implemented with `fs::canonical` (see https://github.com/doxygen/doxygen/blob/a7daf777923a8e1571c317d509609052279924e9/src/fileinfo.cpp#L108) which the former I assume was not. As a consequence after that change the filename used for the markdown content contains the absolute path with symlinks resolved (see https://github.com/doxygen/doxygen/blob/master/src/markdown.cpp#L3467-L3480).

**Proposed fixes**

1. If the semantic change of the `FilePath::absFilePath()` function was unintentional during the mentioned refactoring, change the implementation to use `fs::absolute()` instead of `fs::canonical()`.
2. ~~If changing existing behavior is to be avoided a new function `FilePath::absolute` (or any other name) can be added with the implementation as described in the first option, and the markdown file can use that new function instead. A docblock on the existing function should be added to clearly document that it is returning a canonical path.~~
    * **Update:** another change (see [this comment](https://github.com/doxygen/doxygen/issues/9294#issuecomment-1110007680)) even resolves symlinks at the very beginning, therefore the second option is not sufficient anymore.

I am happy to provide a pull request for either of the options.

**Comment by albert-github** [comment]
Might be related to #9211

**Comment by dirk-thomas** [comment]
There is a second change which resolves symlinks even earlier in the process: https://github.com/doxygen/doxygen/commit/0d05e79d6#diff-2febfc8b9bd51f32694ec80f19df8c447374a21342e56b92c309772f7bc0f413R9750

That resolves the symlinks even before passing the file path to the markdown API.

**Comment by doxygen** [comment]
@dirk-thomas Please verify if commit https://github.com/doxygen/doxygen/commit/62bea90229c56e71d1fe150cb4b19d5e4fc6e9e0 fixes the problem.

**Comment by dirk-thomas** [comment]
I can confirm that this commit fixes the problem described in this ticket. Thank you very much for addressing it so quickly :bow:

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.4.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
